### PR TITLE
virt-manager: support building on darwin

### PR DESCRIPTION
**Pull request by basvandijk**
###### Motivation for this change

This enables building and running `virt-manager` on darwin. 

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by basvandijk** [comment]
@GrahamcOfBorg build virt-manager

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 113306` at b88422cc run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li><a href="https://gist.github.com/05688018ec05472e3da1eee2223191c0">gnome3.gnome-boxes</a></li></li>
    <li><a href="https://gist.github.com/e89f7b4dac775204ac6c4d16b31200b7">libvirt-glib</a></li></li>
    <li><a href="https://gist.github.com/865736a24028d1138ecf0fd2e3979c3b">virt-manager</a></li></li>
    <li><a href="https://gist.github.com/73788c178df876c42afeffe2651a15e4">virt-viewer</a></li></li>
  </ul>
</details>
<details>
  <summary>2 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md">maintainers-missing</a>

Package does not have a maintainer. Consider adding yourself?

Near pkgs/development/libraries/libvirt-glib/default.nix:25:3:
```
   |
25 |   meta = with lib; {
   |   ^
```
</li>
    <li>warning: unused-argument

Unused argument: `glib`.
Near pkgs/applications/virtualization/virt-manager/default.nix:4:30:
```
  |
4 | , gsettings-desktop-schemas, glib, libosinfo, gnome3
  |                              ^
```
</li>
  </ul>
</details>

File: pkgs/applications/virtualization/virt-manager/default.nix [hunk 1]
```diff
@@ -1,9 +1,9 @@
-{ lib, fetchurl, python3Packages, intltool, file
+{ lib, stdenv, fetchurl, python3Packages, intltool, file
 , wrapGAppsHook, gtk-vnc, vte, avahi, dconf
 , gobject-introspection, libvirt-glib, system-libvirt
 , gsettings-desktop-schemas, glib, libosinfo, gnome3
```
Comment (SuperSandro2000) [inline pkgs/applications/virtualization/virt-manager/default.nix:4]:
```suggestion
, gsettings-desktop-schemas, glib, libosinfo, gnome3
```
```suggestion
, gsettings-desktop-schemas, libosinfo, gnome3
```

File: pkgs/applications/virtualization/virt-manager/default.nix [hunk 2]
```diff
@@ -1,9 +1,9 @@
...
 , gtksourceview4, docutils
-, spiceSupport ? true, spice-gtk ? null
+, spiceSupport ? !stdenv.isDarwin, spice-gtk ? null
```
Comment (SuperSandro2000) [inline pkgs/applications/virtualization/virt-manager/default.nix:6]:
```suggestion
, spiceSupport ? !stdenv.isDarwin, spice-gtk
```

File: pkgs/applications/virtualization/virt-manager/default.nix [hunk 3]
```diff
@@ -26,10 +26,12 @@ python3Packages.buildPythonApplication rec {
...
   buildInputs = [
     wrapGAppsHook
-    libvirt-glib vte dconf gtk-vnc gnome3.adwaita-icon-theme avahi
+    libvirt-glib dconf gnome3.adwaita-icon-theme avahi
     gsettings-desktop-schemas libosinfo gtksourceview4
     gobject-introspection # Temporary fix, see https://github.com/NixOS/nixpkgs/issues/56943
-  ] ++ optional spiceSupport spice-gtk;
+  ]
+  ++ optional spiceSupport spice-gtk
+  ++ optionals (!stdenv.isDarwin) [ gtk-vnc vte ];
```
Comment (SuperSandro2000) [inline pkgs/applications/virtualization/virt-manager/default.nix:34]:
```suggestion
  ] ++ optional spiceSupport spice-gtk
  ++ optionals (!stdenv.isDarwin) [ gtk-vnc vte ];
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 113306` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
4 packages built:
</summary>
<ul>
<li>gnome3.gnome-boxes</li>
<li>libvirt-glib</li>
<li>virt-manager</li>
<li>virt-viewer</li>
</ul>
</details>

The following issues got detected with the above build packages.
Please fix at least the ones listed with your changed packages:

gnome3.gnome-boxes:
<details>


`gpl3` is a deprecated license, check if project uses `gpl3Plus` or `gpl3Only` and change `meta.license` accordingly.

Near pkgs/desktops/gnome-3/apps/gnome-boxes/default.nix:141:5:

```
    |
141 |     license = licenses.gpl3;
    |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md
</details>
libvirt-glib:
<details>


Package does not have a maintainer. Consider adding yourself?

Near pkgs/development/libraries/libvirt-glib/default.nix:25:3:

```
   |
25 |   meta = with lib; {
   |   ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md
gobject-introspection is a build tool so it likely goes to `nativeBuildInputs`, not `buildInputs`.

Near pkgs/development/libraries/libvirt-glib/default.nix:19:3:

```
   |
19 |   buildInputs = [ libvirt libxml2 gobject-introspection ]
   |   ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/build-tools-in-build-inputs.md
</details>
virt-manager:
<details>


Unused argument: `glib`.
Near pkgs/applications/virtualization/virt-manager/default.nix:4:30:

```
  |
4 | , gsettings-desktop-schemas, glib, libosinfo, gnome3
  |                              ^
```
</details>
virt-viewer:
<details>


`gpl2` is a deprecated license, check if project uses `gpl2Plus` or `gpl2Only` and change `meta.license` accordingly.

Near pkgs/applications/virtualization/virt-viewer/default.nix:39:5:

```
   |
39 |     license = licenses.gpl2;
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md
</details>

**Comment by siraben** [comment]
Please resolve the merge conflict.

**Comment by mikroskeem** [comment]
Any updates on this?

**Comment by AndersonTorres** [comment]
Closing because https://github.com/NixOS/nixpkgs/pull/152350 will supercede this.

**Pull request closed without merging**